scipy/stats/_hypotests.py

```diff
@@ -663,3 +666,352 @@ def somersd(x, y=None):
         raise ValueError("x must be either a 1D or 2D array")
     d, p = _somers_d(table)
     return SomersDResult(d, p, table)
+
+
+def _compute_log_combinations(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = gammaln(np.arange(n + 1) + 1)
+    return gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+BarnardExactResult = make_dataclass(
+    "BarnardExactResult", [("statistic", float), ("pvalue", float)]
+)
+
+
+def barnard_exact(table, alternative="two-sided", pooled=True, shgo_n=32):
+    r"""Perform a Barnard exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+        * 'two-sided'
+        * 'less': one-sided
+        * 'greater': one-sided
+
+    pooled : bool, optional
+        Whether to compute score statistic with pooled variance (Student's
+        t-test) or unpooled variance (Welch's t-test). Default is ``True`` :
+        Statistic test is computed using pooled variance.
+
+    shgo_n : int, optional
+        Number of sampling points used in the construction of the sobol
```

Comment:
Or alternatively the method so a reference to ```  `scipy.stats.qmc.Sobol` ```
```suggestion
        Number of sampling points used in the construction of the Sobol'
```

Reply:
I have updated the doc in aebd2b88e71c4e47a4455992fc2707dc76b2b2f2. I was not sure what you wanted exactly so I improvised 😆Is it what you wanted?